### PR TITLE
ORC-787: Update site to use `main`

### PR DESCRIPTION
**Pull request by pgaref**
### What changes were proposed in this pull request?
Update site to use `main`

### Why are the changes needed?
Change needed after INFRA-21723

### How was this patch tested?
Locally build and run site

### CODE REVIEW

**Review by wgtmac — APPROVED** [review]

File: site/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ the site is to use docker to use a standard environment.
 ## Setup
 
 1. `cd site`
-2. `git clone git@github.com:apache/orc.git -b asf-site target`
```
Comment (omalley) [inline site/README.md:10]: Since this is for developers, I'd think that the ssh version is better.

**Review by omalley — CHANGES_REQUESTED** [review]

File: site/README.md (hunk 1 above)
Comment (pgaref) [reply site/README.md]: Sure, updated

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for doing this.

**Comment by dongjoon-hyun** [comment]
Do you have any other comments, @omalley ?

**Comment by dongjoon-hyun** [comment]
Since all comments are addressed, I'll merge this. Thank you all!

**Pull request merged**